src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -42,10 +52,73 @@ public List<IexSymbol> getAllStockSymbols() {
    * @return a List of IexLastTradedPrice objects for the given symbols.
    */
   @GetMapping(value = "${mvc.iex.getLastTradedPricePath}", produces = {
-      MediaType.APPLICATION_JSON_VALUE})
+          MediaType.APPLICATION_JSON_VALUE})
   public List<IexLastTradedPrice> getLastTradedPrice(
-      @RequestParam(value = "symbols") final List<String> symbols) {
+          @RequestParam(value = "symbols") final List<String> symbols) {
     return iexService.getLastTradedPriceForSymbols(symbols);
   }
 
+  /**
+   * Get historical pricing data for the given symbol on the given date
+   *
+   * @param symbols the symbol to retrieve data about
+   * @param dates   the date which should be queried
+   * @return historical pricing data for the given symbol on the given date
+   */
+  @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
+          MediaType.APPLICATION_JSON_VALUE})
+  public List<IexHistoricalPrice> getHistoricalPrice(
+          @RequestParam(value = "symbol", required = true) final List<String> symbols,
+          @RequestParam(value = "range", required = false) final List<String> ranges,
+          @RequestParam(value = "date", required = false) @DateTimeFormat(pattern = "yyyyMMdd") final List<LocalDate> dates) {
+    //date mode
+    if ((symbols.size() == 1) && (dates != null) && (dates.size() == 1) && (ranges == null)) {
+      LocalDate date = dates.get(0);
+
+      if (checkValidDate(date)) {
+        return iexService.getHistoricalPriceOnDate(symbols.get(0), dates.get(0));
+      }
+      else {
+        return Collections.emptyList();
```

Comment:
Generally, it's better to return an error if something happens. Returning an empty response is ambiguous for the user, because it's hard to tell whether the empty response is due to no data or due to an error.

There may be some cases in which we want to do it anyway, due to certain requirements we might have. But for this endpoint let's say we do want to return an error. The easiest way to do that is to just throw an exception. There are ways we can customize the response associated with certain exceptions, so try to do that here - we want a 400 Bad Request response code here, rather than a 500 Internal Server Error response (which is the default status code for most exceptions).